### ISSUE TITLE
pip can't build wheel with LibreSSL 3.6.0

### ISSUE THREAD

**Issue by jpcw** [pyca/cryptography#7807]
* Versions of Python, ``cryptography``, ``cffi``, ``pip``, and ``setuptools``
  you're using

- platform: OpenBSD-7.2-amd64-64bit-ELF
- **libressl 3.6**
- python 3.9.15
- pip 22.3.1
- cffi 1.15.1
- cryptography 38.0.3
- setuptools: 65.5.1
- setuptools_rust: 1.5.2
- rustc: 1.65.0


* How you installed ``cryptography``
* Clear steps for reproducing your bug

```
python3 -m venv bug_crypto
cd bug_crypto/
bin/pip install pip --upgrade
bin/pip install cryptography
```

The error here:
```
      Copying rust artifact from /tmp/pip-install-c59utlqj/cryptography_15f20c3083684126bc25cfaa79cf6422/src/rust/target/release/libcryptography_rust.so to build/lib.openbsd-7.2-amd64-cpython-39/cryptography/hazmat/bindings/_rust.abi3.so                                    
      building 'cryptography.hazmat.bindings._openssl' extension                                                                                                                                                                                                                 
      creating build/temp.openbsd-7.2-amd64-cpython-39/build                                                                                                                                                                                                                     
      creating build/temp.openbsd-7.2-amd64-cpython-39/build/temp.openbsd-7.2-amd64-cpython-39                                                                                                                                                                                   
      cc -pthread -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -O2 -pipe -g -fPIC -O2 -pipe -g -O2 -pipe -g -fPIC -I/home/jpcw/bug_crypto/include -I/usr/local/include/python3.9 -c build/temp.openbsd-7.2-amd64-cpython-39/cryptography.hazmat.bindings._openss
l.c -o build/temp.openbsd-7.2-amd64-cpython-39/build/temp.openbsd-7.2-amd64-cpython-39/cryptography.hazmat.bindings._openssl.o -Wconversion -Wno-error=sign-conversion                                                                                                           
      build/temp.openbsd-7.2-amd64-cpython-39/cryptography.hazmat.bindings._openssl.c:961:8: error: redefinition of 'OPENSSL_cleanup' as different kind of symbol                                                                                                                
      void (*OPENSSL_cleanup)(void) = NULL;                                                                                                                                                                                                                                      
             ^                                                                                                                                                                                                                                                                   
      /usr/include/openssl/crypto.h:565:6: note: previous definition is here                                                                                                                                                                                                     
      void OPENSSL_cleanup(void);                                                                                                                                                                                                                                                
           ^                      
```

Notice : OpenBSD ports maintainers commit a patch to build : py3-cryptography-38.0.3p0

Seems the the following patch solves the "OPENSSL_cleanup" problem

https://github.com/openbsd/ports/commit/8c2bfe49f1d114455b417e380e84608e8303cdb1

Thanks

**Comment by alex** [comment]
This issue is resolved on main, but the fix is not yet included in a
release.

**Comment by jpcw** [comment]
Thanks @alex 
Will wait the release.

**Comment by alex** [comment]
This fix is included in the 38.0.4 release.
